addons/portal/static/src/js/portal.js

```diff
@@ -90,7 +93,17 @@ publicWidget.registry.PortalHomeCounters = publicWidget.Widget.extend({
                 Object.keys(data).map(k => this.$("[data-placeholder_count='" + k + "']").text(data[k]));
             });
         });
-        return Promise.all(proms);
+        return Promise.all(proms).then(() => {
+            this.$('.o_portal_doc_spinner').remove();
+            this.$('[data-placeholder_count]').toArray().forEach(counter => {
+                if (counter.innerText !== '0' || this.countersToKeep.includes(counter.dataset['placeholder_count'])) {
+                    $(counter.closest('a')).removeClass('d-none');
+                }
+            });
+            if (!this.$('[data-placeholder_count]').parent().not('.d-none').length) {
```

Comment:
Is this "we don't have anything at all to display" ?

Can't it rely on counter results from rpc directly instead of inferring this from a d-none class?